### DISCUSSION TITLE
Automatically Not Be In Normal Mode In Read Only Buffers

### DISCUSSION THREAD

**Discussion by jgarte** [meow-edit/meow#486] [Q&A] [answered]
Hi,

How can I have meow automatically switch to insert mode when in a read only buffer.

For example,

I get binding conflicts when in a [sly](https://joaotavora.github.io/sly/) debugger or inspector.

**Comment by DogLooksGood** [comment, marked as answer]
I think you have to switch to motion state by hook or advice when entering the debug mode.

There are examples,
https://github.com/meow-edit/meow/blob/master/meow-shims.el#L220
https://github.com/meow-edit/meow/blob/master/meow-shims.el#L199

BTW, PR is welcomed.

**Reply by jgarte** [reply]
Will attempt soon. Thanks!

**Comment by jgarte** [comment]
Hi,

Should I be adding advice around `meow--enable-shims` if adding this shim to my own config?

https://git.sr.ht/~whereiseveryone/emacsaroni/tree/master/item/site-lisp/meow-eat-shim.el

**Reply by DogLooksGood** [reply]
I think you don't have to add advice around `meow--enable-shims`. 
If you think these codes are useful, you can consider a PR as well.
